sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -22,15 +22,16 @@ internal class ClientDiagnostics : DiagnosticScopeFactory
 
         private readonly HttpMessageSanitizer _sanitizer;
 
-        public ClientDiagnostics(ClientOptions options)
+        public ClientDiagnostics(ClientOptions options, bool? suppressNestedClientActivities = null)
                     : this(options.GetType().Namespace!,
                     GetResourceProviderNamespace(options.GetType().Assembly),
-                    options.Diagnostics)
+                    options.Diagnostics,
+                    suppressNestedClientActivities)
         {
         }
 
-        public ClientDiagnostics(string optionsNamespace, string? providerNamespace, DiagnosticsOptions diagnosticsOptions)
-            : base(optionsNamespace, providerNamespace, diagnosticsOptions.IsDistributedTracingEnabled, false)
+        public ClientDiagnostics(string optionsNamespace, string? providerNamespace, DiagnosticsOptions diagnosticsOptions, bool? suppressNestedClientActivities = null)
```

Comment:
Why not `bool suppressNestedClientActivities = false` so callers know the default behavior more easily?

Reply:
+1, curious about the reasoning behind having it be nullable?

Reply:
the reasoning is to be able to change the default once we're ready. So we need three choices:
1. `true`: clients explicitly opting-into suppression (new ones) 
2. `false`: clients that explicitly opt-out. E.g. HTTP RequestActivityPolicy that must never be suppressed
3. `null`: clients that follow default (current SDKs) - that only want back-compat

If we merge `null` with `false` we'll have a very hard time separating explicit choice from following a default.

Reply:
Would you mind adding a comment explaining the reasoning behind why this is nullable?

Reply:
Oh, that is cool. 😎